### ISSUE TITLE
Update won't work when number is longer than previous number added

### ISSUE THREAD

**Issue by TylexTy** [TylexTy/fullstackopen#1]
https://github.com/TylexTy/fullstackopen/blob/0669ea9055717a6258fc9a8bcfe6816c7447f465/part2/phonebook/src/App.js#L68

**Comment by ryanlozon** [comment]
It would be helpful to paste the entire error message here.

**Comment by ryanlozon** [comment]
So when you compare 'person.name.localeCompare(newName) === 0' the it catches an error and says `Information of ${newName} has already been removed from server, e11000 duplicate key error collection`?

**Comment by TylexTy** [comment]
So I deleted the collection in mongoDB and recreated it and it's working better now but not perfect yet. I'm no longer getting the 'Information of...." error but localeCompare is only catching strings that are shorter than the current string and when strings are longer than the current string I get 'user is already added to phonebook' When I want it to update the number

**Comment by ryanlozon** [comment]
Would it work to try 'person.name === newName'? I'm not really familiar with localeCompare but it might be looking at substrings.

**Comment by TylexTy** [comment]
yeah that works thanks. I didn't think you could compare strings with the equals signs in javascript. I guess there aren't pointers in javascript to worry about though. Now it's updating but it's updating the wrong contact

**Comment by TylexTy** [comment]
Just pushed the updated code. It's not working atm, I broken it again. I'm going to take this opportunity to learn how to debug, and after this I'm going to start using git flow method

**Comment by ryanlozon** [comment]
Hmm... I wonder why it would be updating the wrong contact. What do you get when you print person.name and newName?

**Comment by TylexTy** [comment]
I think it might be because I had 2 id's? earlier in the course we generated a random id, but now that we are using mongoDB we have an id created for us. I took out the id in schema and other places and now it's back to giving me the error that we started with, about duplicate name_1 when I try to update. I'm part of a telegram community with this course, might see if someone can help me there, I feel in way over my head with this.

**Comment by ryanlozon** [comment]
Well, I think the property for mongo's Id is actually "_id", you could try that out.

**Comment by ryanlozon** [comment]
I would also just do your find one and save it in a variable. It would save computing multiple iterations. 
i.e const contactExists = persons.find(person => person.name.toLowerCase() === newName.toLowerCase()

Putting comments over your if/else segments can also really help readability. 

But really good stuff man! :)

**Comment by TylexTy** [comment]
true, yeah I'll add comments, I'm always bad with that, thanks for the help man!
